File: lib-mvpbase-appcompat/src/main/java/com/baozi/mvp/templet/options/ToolbarOptions.java
package com.baozi.mvp.templet.options;

import android.support.annotation.ColorInt;
import android.support.annotation.DrawableRes;
import android.support.annotation.LayoutRes;

import com.baozi.mvp.templet.helper.ToolbarHelper;

/**
 * Created by baozi on 2017/10/25.
 */

public class ToolbarOptions implements Cloneable {
    //其他字体大小,颜色
    private int otherTextSize;
    @ColorInt
    private int otherTextColor;
    //标题字体大小,颜色
    private int titleSize;
    @ColorInt
    private int titleColor;
    //toolbar配置
    private int toolbarHeight;
    @ColorInt
    private int toolbarColor;
    @LayoutRes
    private int toolbarLayout = ToolbarHelper.TOOLBAR_TEMPLET_DEFUATL;
    @DrawableRes
    private int toolbarDrawable;
    @DrawableRes
    private int statusDrawable;
    //返回图标
    @DrawableRes
    private int backDrawable;

    private boolean noBack;
    private float elevation;

    public static ToolbarOptions Create() {
        return new ToolbarOptions();
    }

    private ToolbarOptions() {

    }

    @DrawableRes
    public int getStatusDrawable() {
        return statusDrawable;
    }

    public ToolbarOptions setStatusDrawable(@DrawableRes int statusDrawable) {
        this.statusDrawable = statusDrawable;
        return this;
    }

    public int getOtherTextSize() {
        return otherTextSize;
    }

    public ToolbarOptions setOtherTextSize(int otherTextSize) {
        this.otherTextSize = otherTextSize;
        return this;
    }

    @ColorInt
    public int getOtherTextColor() {
        return otherTextColor;
    }

    public ToolbarOptions setOtherTextColor(@ColorInt int otherTextColor) {
        this.otherTextColor = otherTextColor;
        return this;
    }

    public int getTitleSize() {
        return titleSize;
    }

    public ToolbarOptions setTitleSize(int titleSize) {
        this.titleSize = titleSize;
        return this;
    }

    @ColorInt
    public int getTitleColor() {
        return titleColor;
    }

    public ToolbarOptions setTitleColor(@ColorInt int titleColor) {
        this.titleColor = titleColor;
        return this;
    }

    public int getToolbarHeight() {
        return toolbarHeight;
    }

    public ToolbarOptions setToolbarHeight(int toolbarHeight) {
        this.toolbarHeight = toolbarHeight;
        return this;
    }

    @ColorInt
    public int getToolbarColor() {
        return toolbarColor;
    }

    public ToolbarOptions setToolbarColor(@ColorInt int toolbarColor) {
        this.toolbarColor = toolbarColor;
        return this;
    }

    @DrawableRes
    public int getToolbarDrawable() {
        return toolbarDrawable;
    }

    public ToolbarOptions setToolbarDrawable(@DrawableRes int toolbarDrawable) {
        this.toolbarDrawable = toolbarDrawable;
        return this;
    }

    @DrawableRes
    public int getBackDrawable() {
        return backDrawable;
    }

    public ToolbarOptions setBackDrawable(@DrawableRes int backDrawable) {
        this.backDrawable = backDrawable;
        return this;
    }

    public boolean isNoBack() {
        return noBack;
    }

    public ToolbarOptions setNoBack(boolean noBack) {
        this.noBack = noBack;
        return this;
    }

    public int getToolbarLayout() {
        return toolbarLayout;
    }

    public ToolbarOptions setToolbarLayout(@LayoutRes int toolbarLayout) {
        this.toolbarLayout = toolbarLayout;
        return this;
    }

    public ToolbarOptions clone() {
        try {
            return (ToolbarOptions) super.clone();
        } catch (CloneNotSupportedException e) {
            throw new RuntimeException(e);
        }
    }

    public ToolbarOptions setElevation(float elevation) {
        this.elevation = elevation;
        return this;
    }

    public float getElevation() {
        return elevation;
    }
}

File: lib-mvpbase-kotlin/src/main/java/com/linfeng/mvp/templet/options/ToolbarOptions.kt
package com.linfeng.mvp.templet.options

import android.support.annotation.ColorInt
import android.support.annotation.DrawableRes
import android.support.annotation.LayoutRes
import com.linfeng.mvp.templet.helper.ToolbarHelper


/**
 * Created by baozi on 2017/10/25.
 */

class ToolbarOptions private constructor() : Cloneable {
    //其他字体大小,颜色
    private var otherTextSize: Int = 0
    @ColorInt
    private var otherTextColor: Int = 0
    //标题字体大小,颜色
    private var titleSize: Int = 0
    @ColorInt
    private var titleColor: Int = 0
    //toolbar配置
    private var toolbarHeight: Int = 0
    @ColorInt
    private var toolbarColor: Int = 0
    @LayoutRes
    private var toolbarLayout = ToolbarHelper.TOOLBAR_TEMPLET_DEFUATL
    @DrawableRes
    private var toolbarDrawable: Int = 0
    @DrawableRes
    private var statusDrawable: Int = 0
    //返回图标
    @DrawableRes
    private var backDrawable: Int = 0

    private var noBack: Boolean = false
    private var elevation: Float = 0.toFloat()

    @DrawableRes
    fun getStatusDrawable(): Int {
        return statusDrawable
    }

    fun setStatusDrawable(@DrawableRes statusDrawable: Int): ToolbarOptions {
        this.statusDrawable = statusDrawable
        return this
    }

    fun getOtherTextSize(): Int {
        return otherTextSize
    }

    fun setOtherTextSize(otherTextSize: Int): ToolbarOptions {
        this.otherTextSize = otherTextSize
        return this
    }

    @ColorInt
    fun getOtherTextColor(): Int {
        return otherTextColor
    }

    fun setOtherTextColor(@ColorInt otherTextColor: Int): ToolbarOptions {
        this.otherTextColor = otherTextColor
        return this
    }

    fun getTitleSize(): Int {
        return titleSize
    }

    fun setTitleSize(titleSize: Int): ToolbarOptions {
        this.titleSize = titleSize
        return this
    }

    @ColorInt
    fun getTitleColor(): Int {
        return titleColor
    }

    fun setTitleColor(@ColorInt titleColor: Int): ToolbarOptions {
        this.titleColor = titleColor
        return this
    }

    fun getToolbarHeight(): Int {
        return toolbarHeight
    }

    fun setToolbarHeight(toolbarHeight: Int): ToolbarOptions {
        this.toolbarHeight = toolbarHeight
        return this
    }

    @ColorInt
    fun getToolbarColor(): Int {
        return toolbarColor
    }

    fun setToolbarColor(@ColorInt toolbarColor: Int): ToolbarOptions {
        this.toolbarColor = toolbarColor
        return this
    }

    @DrawableRes
    fun getToolbarDrawable(): Int {
        return toolbarDrawable
    }

    fun setToolbarDrawable(@DrawableRes toolbarDrawable: Int): ToolbarOptions {
        this.toolbarDrawable = toolbarDrawable
        return this
    }

    @DrawableRes
    fun getBackDrawable(): Int {
        return backDrawable
    }

    fun setBackDrawable(@DrawableRes backDrawable: Int): ToolbarOptions {
        this.backDrawable = backDrawable
        return this
    }

    fun isNoBack(): Boolean {
        return noBack
    }

    fun setNoBack(noBack: Boolean): ToolbarOptions {
        this.noBack = noBack
        return this
    }

    fun getToolbarLayout(): Int {
        return toolbarLayout
    }

    fun setToolbarLayout(@LayoutRes toolbarLayout: Int): ToolbarOptions {
        this.toolbarLayout = toolbarLayout
        return this
    }

    public override fun clone(): ToolbarOptions {
        try {
            return super.clone() as ToolbarOptions
        } catch (e: CloneNotSupportedException) {
            throw RuntimeException(e)
        }

    }

    fun setElevation(elevation: Float): ToolbarOptions {
        this.elevation = elevation
        return this
    }

    fun getElevation(): Float {
        return elevation
    }

    companion object {

        fun Create(): ToolbarOptions {
            return ToolbarOptions()
        }
    }
}
